src/mono/msbuild/common/LibraryBuilder.targets

```diff
@@ -35,4 +41,35 @@
     </LibraryBuilderTask>
   </Target>
 
+  <Target Name="_GenerateBundle"
+          Condition="'$(BundlesAssemblies)' == 'true'">
+
+    <PropertyGroup>
+      <_AssembliesToBundleSourceFile>$(BundleDir)bundled_assemblies.c</_AssembliesToBundleSourceFile>
+    </PropertyGroup>
+
+    <ItemGroup>
+      <_AssembliesToBundle Remove="@(_AssembliesToBundle)" />
+      <_AssembliesToBundle Include="@(_AssembliesToBundleInternal)" />
```

Comment:
I'm pretty sure we will end up in scenario where user want to select what assets to bundle into the library and what to keep on disk, so then we would need to have metadata on items list that we filter on and pass to bundle task, or the bundle task evaluate the metadata. We have similar concept on other platforms, where each assembly that we run through the toolchain can have metadata set by user. Maybe we can leave it out in this first PR and do it follow up PR.

We already know of one scenario where this might be needed, on maccatalyst you can create on framework that includes several different target platforms, so you end up with libraries targeting x64/arm64 in the same framework, but since System.Private.CoreLib is tied close to the target platform, you would like to bundle that assembly into the library, but keep other assemblies on disk so they can be shared between platforms.